cat-test/tests/cat_test.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import 'package:cat_test/cat_cafe.dart';
+import 'package:cat_test/cat.dart';
+
+@proxy
+class MockCatCafe extends Mock implements CatCafe {}
+
+class MockCat extends Mock implements Cat {}
+
+main() {
+  test('Cat.getFood() returns food from the cat cafe', () {
+    var cafe = new MockCatCafe();
+    var cat = new Cat(cafe);
+
+    when(cafe.serveMeal()).thenReturn('kibble');
+    expect(cat.getFood(), 'kibble');
```

Comment:
I'd end the simplest example here. Just one expect. It's also a recommended best practice by some people (not me) to stay with only one expect per test case.

Another file (maybe catcafe_test, testing the other way around) can show `verify().called()`.